src/client/common/installer/pipEnvInstaller.ts

```diff
@@ -29,11 +28,18 @@ export class PipEnvInstaller extends ModuleInstaller {
 
     constructor(@inject(IServiceContainer) serviceContainer: IServiceContainer) {
         super(serviceContainer);
-        this.pipenv = this.serviceContainer.get<IInterpreterLocatorService>(IInterpreterLocatorService, PIPENV_SERVICE);
     }
     public async isSupported(resource?: InterpreterUri): Promise<boolean> {
         if (isResource(resource)) {
-            const interpreters = await this.pipenv.getInterpreters(resource);
+            const experimentService = this.serviceContainer.get<IExperimentService>(IExperimentService);
+            const pyenvs = this.serviceContainer.get<IComponentAdapter>(IComponentAdapter);
+            const interpreters = (await inDiscoveryExperiment(experimentService))
+                ? await pyenvs
+                      .getInterpreters(resource)
+                      .then((envs) => envs.filter((e) => e.envType == EnvironmentType.Pipenv))
```

Comment:
Why not add an explicit method, like we have in the other cases?  Then this fits into the pattern we've been following:

```suggestion
                ? await pyenvs.getPipenvInterpreters(resource)
```

Reply:
My idea was to not create new methods in component adapter, which is already bloated, if we can use existing methods. Other cases where I created methods: `getWorkspaceEnvs`, it was not possible to use existing methods.

Eventually when we get rid of component adapter I suspect we'll only have one method with additional query options, like we have with the API.

Reply:
Yeah, `ComponentAdapter` is only a temporary hack.  So it is better to add all the simple methods we need to that match the old code.

Reply:
I'm going more towards what the code will eventually look like, where we won't have separate methods for each type of query. The existing code is this PR is closer to that.

Reply:
Eventually we will use only the `PythonEnvironments` object.  We're not very close to that, so let's keep things simple with `ComponentAdapter`.

Reply:
I'm not going to block this PR on this, but note that I'm uncomfortable with adding generic stuff to `ComponentAdapter` when the original code was not generic.